File: python/P024_Swap_Nodes_in_Pairs/SwapNodesinPairs.py
# -*- coding: utf-8 -*-

# Definition for singly-linked list.
class ListNode(object):
    def __init__(self, x):
        self.val = x
        self.next = None


class Solution(object):
    def swapPairs(self, head):
        if head and head.next:
            next, temp = head.next, head.next.next
            next.next = head
            head.next = self.swapPairs(temp)
            return next
        return head

File: java/src/P024/SwapNodesinPairs/SwapNodesinPairs.java
package P024.SwapNodesinPairs;

public class SwapNodesinPairs {
    public ListNode swapPairs(ListNode head) {
        if (head != null && head.next != null) {
            ListNode next = head.next;
            ListNode temp = next.next;

            next.next = head;
            head.next = this.swapPairs(temp);

            return next;
        }
        return head;
    }
}
